File: com-223 Data Structures/2017-2018 Data Structures/Lab07/Rect.cpp
#include <sstream>
#include "Rect.h"

using namespace std;

Rect::Rect(int aX, int aY, int aW, int aH)
: AbstractFigure(aX, aY), w(aW), h(aH)
{
}

string Rect::toString() const
{
    ostringstream sout;
    sout << "Rect[" << x << ", " << y << ", " << w << ", " << h << "]";
    
    return sout.str();
}

bool Rect::contains(int aX, int aY) const
{
    int bx = x <= aX and aX <= x + w;
    int by = y <= aY and aY <= y + h;
    
    return bx and by;
}

File: com-117 Programming2/Programming02/Lab06Problem01/src/Rect.java
import java.awt.*;

public class Rect extends Figure {
	
	public Rect(int x, int y, int w, int h) {
		super(x,  y);
		this.w = w;
		this.h = h;
	}

	@Override
	public void draw(Graphics g) {
		g.setColor(Color.RED);
		g.fillRect(x, y, w, h);
	}

	@Override
	public boolean contains(int x, int y) {
		boolean b1 = this.x <= x && x <= this.x + this.w;
		boolean b2 = this.y <= y && y <= this.y + this.h;
		return b1 && b2;
	}
	
	@Override
	public String toString() {
		return String.format("Rect: x=%d, y=%d, w=%d, h=%d", x, y, w, h);
	}
}
